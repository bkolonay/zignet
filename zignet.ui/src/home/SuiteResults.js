import React, { Component } from 'react';
import SuiteResultChart from './SuiteResultChart'
import GroupSuitesButton from './GroupSuitesButton'

class SuiteResults extends Component {
  render() {
    return (
  	  <div className="container">
        <div className="text-right">
          <GroupSuitesButton />
        </div>
  	  	<div className="row">
          {this.props.suiteResults.map((suiteResult) =>
            <SuiteResultChart key={suiteResult.SuiteIds[0]}
                              suiteResult={suiteResult} />
          )}
        </div>
      </div>
    );
  }
}

export default SuiteResults;
