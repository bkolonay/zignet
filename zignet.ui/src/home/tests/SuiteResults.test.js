import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SuiteResults from '../SuiteResults';
jest.mock('react-router-dom');

it('renders single suite result', () => {
  const suiteResults = [{
    'SuiteIds':[1],
    'SuiteName':'suite-name',
    'TotalPassedTests':1,
    'TotalFailedTests':10,
    'TotalInconclusiveTests':0,
    'SuiteEndTime':'2018-02-28T15:05:00'
  }];
  
  const div = document.createElement('div');
  ReactDOM.render(
    <SuiteResults suiteResults={suiteResults} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders zero suite results', () => {
  const suiteResults = [];
  
  ReactDOM.render(
    <SuiteResults suiteResults={suiteResults} />,
    document.createElement('div')
  );
});

it('renders multiple suite results', () => {
  const suiteResults = [
    {
      'SuiteIds':[1],
      'SuiteName':'suite-name',
      'TotalPassedTests':1,
      'TotalFailedTests':10,
      'TotalInconclusiveTests':0,
      'SuiteEndTime':'2018-02-28T15:05:00'
    },
    {
      'SuiteIds':[2],
      'SuiteName':'suite-name 2',
      'TotalPassedTests':9,
      'TotalFailedTests':24,
      'TotalInconclusiveTests':3,
      'SuiteEndTime':'2019-03-28T16:19:00'
    }
  ];
  
  const div = document.createElement('div');
  ReactDOM.render(
    <SuiteResults suiteResults={suiteResults} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the previous snapshot', () => {
  const suiteResults = [
    {
      'SuiteIds':[1],
      'SuiteName':'suite-name',
      'TotalPassedTests':1,
      'TotalFailedTests':10,
      'TotalInconclusiveTests':0,
      'SuiteEndTime':'2018-02-28T15:05:00'
    },
    {
      'SuiteIds':[2],
      'SuiteName':'suite-name 2',
      'TotalPassedTests':9,
      'TotalFailedTests':24,
      'TotalInconclusiveTests':3,
      'SuiteEndTime':'2019-03-28T16:19:00'
    }
  ];

  const tree = renderer
    .create(<SuiteResults suiteResults={suiteResults} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});