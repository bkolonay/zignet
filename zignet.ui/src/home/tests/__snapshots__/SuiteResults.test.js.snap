// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with multiple suites 1`] = `
<div
  className="container"
>
  <div
    className="text-right"
  >
    <a
      className="btn btn-primary"
      href="/?group=true"
      role="button"
      title="Click to group results by environment"
    >
      Group
    </a>
  </div>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h3
        className="text-center"
      >
        suite-name
      </h3>
      <div
        className="ct-chart"
        id="chart1"
      />
      <p
        className="text-center chart-label"
      >
        <a
          href="/1"
        >
          Total: 
          110
        </a>
      </p>
      <p
        className="text-center text-muted"
      >
        <small>
          3 months ago
        </small>
      </p>
    </div>
    <div
      className="col-4"
    >
      <h3
        className="text-center"
      >
        second-suite-name
      </h3>
      <div
        className="ct-chart"
        id="chart2"
      />
      <p
        className="text-center chart-label"
      >
        <a
          href="/2"
        >
          Total: 
          100
        </a>
      </p>
      <p
        className="text-center text-muted"
      >
        <small>
          in 10 months
        </small>
      </p>
    </div>
  </div>
</div>
`;

exports[`snapshot with single grouped suite 1`] = `
<div
  className="container"
>
  <div
    className="text-right"
  >
    <a
      className="btn btn-outline-primary"
      href="/"
      role="button"
      title="Click to ungroup results"
    >
      Ungroup
    </a>
  </div>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h3
        className="text-center"
      >
        grouped-suite-name
      </h3>
      <div
        className="ct-chart"
        id="chart1"
      />
      <p
        className="text-center chart-label"
      >
        <a
          href="/1?group=true"
        >
          Total: 
          300
        </a>
      </p>
      <p />
    </div>
  </div>
</div>
`;

exports[`snapshot with single suite 1`] = `
<div
  className="container"
>
  <div
    className="text-right"
  >
    <a
      className="btn btn-primary"
      href="/?group=true"
      role="button"
      title="Click to group results by environment"
    >
      Group
    </a>
  </div>
  <div
    className="row"
  >
    <div
      className="col-4"
    >
      <h3
        className="text-center"
      >
        suite-name
      </h3>
      <div
        className="ct-chart"
        id="chart1"
      />
      <p
        className="text-center chart-label"
      >
        <a
          href="/1"
        >
          Total: 
          110
        </a>
      </p>
      <p
        className="text-center text-warning"
      >
        <small>
          running...
        </small>
      </p>
    </div>
  </div>
</div>
`;

exports[`snapshot with zero suites 1`] = `
<div
  className="container"
>
  <div
    className="text-right"
  >
    <a
      className="btn btn-primary"
      href="/?group=true"
      role="button"
      title="Click to group results by environment"
    >
      Group
    </a>
  </div>
  <div
    className="row"
  />
</div>
`;
